### ISSUE TITLE
Creating a static thrift table with integer comparator fails

### ISSUE THREAD

**Issue by tgrabiec** [scylladb/scylla#2571]
*Installation details*
Scylla version (or git commit hash): 2.0, master

dtest, in thrift_test.py:

```python
def test_nonstandard_comparator_with_column_metadata(self):
        ks = 'Keyspace1'
        _set_keyspace(ks)
        cd = ColumnDef(_i64(4), 'LongType', None, None)
        cf = CfDef('Keyspace1', 'test_table', column_metadata=[cd], comparator_type='LongType')
        client.system_add_column_family(cf)
```

Passes with Cassandra, fails with Scylla:

```
======================================================================
ERROR: test_nonstandard_comparator_with_column_metadata (thrift_tests.TestMutations)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tgrabiec/src/scylla-dtest/thrift_tests.py", line 1802, in test_nonstandard_comparator_with_column_metadata
    client.system_add_column_family(cf)
  File "/home/tgrabiec/src/scylla-dtest/thrift_bindings/v22/Cassandra.py", line 1758, in system_add_column_family
    return self.recv_system_add_column_family()
  File "/home/tgrabiec/src/scylla-dtest/thrift_bindings/v22/Cassandra.py", line 1781, in recv_system_add_column_family
    raise result.ire
InvalidRequestException: InvalidRequestException(why='marshalling error')
```

```
#0  0x00007ffff7246e9d in __cxa_throw () from /lib64/libstdc++.so.6
#1  0x0000000000ccd137 in integer_type_impl<long>::validate (this=<optimized out>, v=...) at types.cc:178
#2  0x0000000000b87b19 in abstract_type::get_string (this=0x60000048f8c0, b=...) at ./types.hh:519
#3  0x0000000000dc732e in cql3::column_identifier::column_identifier (this=0x6000054729c0, bytes_=..., type=...)
    at cql3/column_identifier.cc:40
#4  0x000000000088a236 in seastar::shared_ptr_count_for<cql3::column_identifier>::shared_ptr_count_for<seastar::basic_sstring<signed char, unsigned int, 31u> const&, seastar::shared_ptr<abstract_type const> > (this=0x6000054729b0)
    at /home/tgrabiec/src/scylla/seastar/core/shared_ptr.hh:337
#5  seastar::shared_ptr_make_helper<cql3::column_identifier, false>::make<seastar::basic_sstring<signed char, unsigned int, 31u> const&, seastar::shared_ptr<abstract_type const> > () at /home/tgrabiec/src/scylla/seastar/core/shared_ptr.hh:500
#6  seastar::make_shared<cql3::column_identifier, seastar::basic_sstring<signed char, unsigned int, 31u> const&, seastar::shared_ptr<abstract_type const> > () at /home/tgrabiec/src/scylla/seastar/core/shared_ptr.hh:518
#7  schema::make_column_specification (this=this@entry=0x7fffffffc1d0, def=...) at schema.cc:104
#8  0x00000000008964b5 in schema::schema (this=0x7fffffffc1d0, raw=..., raw_view_info=std::experimental::optional<raw_view_info> = {...})
    at schema.cc:220
#9  0x0000000000896ddd in schema_builder::build (this=this@entry=0x7fffffffc620) at schema.cc:865
#10 0x0000000000c7865b in thrift_handler::schema_from_thrift (cf_def=..., ks_name=..., 
    id=std::experimental::optional<utils::UUID> [no contained value]) at thrift/handler.cc:1287
#11 0x0000000000c815b8 in thrift_handler::system_add_column_family(std::tr1::function<void (std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)>, std::tr1::function<void (apache::thrift::TDelayedException*)>, cassandra::CfDef const&)::{lambda()#1}::operator()() const (__closure=0x7fffffffc940) at thrift/handler.cc:806
#12 0x0000000000c81880 in void with_cob<thrift_handler::system_add_column_family(std::tr1::function<void (std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)>, std::tr1::function<void (apache::thrift::TDelayedException*)>, cassandra::CfDef const&)::{lambda()#1}, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >(std::tr1::function<void (std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)>&&, std::tr1::function<void (apache::thrift::TDelayedException*)>&&, thrift_handler::system_add_column_family(std::tr1::function<void (std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)>, std::tr1::function<void (apache::thrift::TDelayedException*)>, cassandra::CfDef const&)::{lambda()#1}&&)::{lambda()#1}::operator()() const (__closure=0x7fffffffc940) at thrift/handler.cc:126
#13 seastar::futurize<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >::apply<void with_cob<thrift_handler::system_add_column_family(std::tr1::function<void (std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)>, std::tr1::function<void (apache::thrift::TDelayedException*)>, cassandra::CfDef const&)::{lambda()#1}, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >(std::tr1::function<void (std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)>&&, std::tr1::function<void (apache::thrift::TDelayedException*)>&&, thrift_handler::system_add_column_family(std::tr1::function<void (std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)>, std::tr1::function<void (apache::thrift::TDelayedException*)>, cassandra::CfDef const&)::{lambda()#1}&&)::{lambda()#1}>(void with_cob<thrift_handler::system_add_column_family(std::tr1::function<void (std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)>, std::tr1::function<void (apache::thrift::TDelayedException*)>, cassandra::CfDef const&)::{lambda()#1}, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >(std::tr1::function<void (std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)>&&, std::tr1::function<void (apache::thrift::TDelayedException*)>&&, thrift_handler::system_add_column_family(std::tr1::function<void (std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)>, std::tr1::function<void (apache::thrift::TDelayedException*)>, cassandra::CfDef const&)::{lambda()#1}&&)::{lambda()#1}) (func=...) at /home/tgrabiec/src/scylla/seastar/core/future.hh:1236
```

Exception happens because `_regular_column_name` is set to `long_type` (fine with the old layout, not with the new layout), which fails to validate the new synthetic regular column named `"value"`.

Refs #2555

**Comment by duarten** [comment]
This is fixed by f685f7f8a18a18a7ba93096a5338e72e244a0e52, no?

**Comment by tgrabiec** [comment]
Right
